client/components/search/index.jsx

```diff
@@ -93,16 +92,17 @@ class Search extends Component {
 		hideOpenIcon: false,
 	};
 
-	state = {
-		keyword: this.props.initialValue || '',
-		isOpen: !! this.props.isOpen,
-		hasFocus: false,
-	};
+	constructor( props ) {
+		super( props );
 
-	componentWillMount() {
-		this.setState( {
-			instanceId: ++Search.instances,
-		} );
+		instanceCount++;
```

Comment:
Careful when refactoring base components, this opens up the regression testing needed for this PR. 

It's usually easier to pull out changes like this to it's own PR so we can clearly explain why it's needed and verify that other search instances have not regressed.